### PR TITLE
Fix bug python38

### LINKED ISSUE #38: Test setup should cover all supported Python versions
Currently the Tox tests are only run against Python 3.6. #37 and #36 both show Python 3.8 issues that could be cause with automated test coverage.

Proposal:
* We create a policy of what Python versions are supported. I suggest we follow upstream HomeAssistant policy of [supporting the 2 most recent minor stable releases](https://github.com/home-assistant/architecture/blob/master/adr/0002-minimum-supported-python-version.md).
* Extend Tox coverage to test with both these supported versions both for local testing and for the automated testing

Immediate consequences:
* Python 3.6 would be deprecated and heading for removal in the next couple releases.

### PR DESCRIPTION
**Pull request by FlorianKempenich**
This pull request fixes #38 and implement a much cleaner way of mocking the arguments passed via the appdaemon config (accessible via the field `args` in any automation)

### CODE REVIEW

**Comment by FlorianKempenich** [comment]
**Update: This breaks compatibility with python 3.6**

As agreed in #38 this is fine, but we will need to remember to do a MAJOR release when bumping the version. 
On another aspect, it also broke the travis build. I'm trying to figure out why and how to fix it. We might see a lot of "Try to fix Travis build" commits as . . . I don't know how to fix a CD problem without actually pushing to the CD system 😅 (if there's a way, I'd love to know)

**Comment by FlorianKempenich** [comment]
Ok, so I renamed this as a WIP. Because it's a bit more complex than I expected. It fails on python 3.8 on travis, but works fine locally :/ I'll look into it

**Comment by FlorianKempenich** [comment]
Ok, travis is passing and testing both python 3.7 and 3.8 😃

File: appdaemontestframework/hass_mocks.py [hunk 1]
```diff
@@ -136,23 +135,66 @@ def _log_log(msg, level='INFO'):
...
+        super().__init__(object_to_patch, field_name)
+
+    def _patch_kwargs(self, _side_effect, _autospec):
+        return {
+            'create': True,
+            'new': self.MockDict()
+        }
+
+
+class SpyMockHandler(MockHandler):
```
Comment (lifeisafractal) [inline appdaemontestframework/hass_mocks.py]: Much better name for this 👍

Comment (FlorianKempenich) [reply appdaemontestframework/hass_mocks.py]: 😊

Comment (FlorianKempenich) [reply appdaemontestframework/hass_mocks.py]: Wait I just realized ... we don't actually use it anywhere, do we? 😅 (`SpyMockHandler`)

It's probably a leftover from when you split the large PR? If so it's cool, let's keep it until you use it, and if not, delete it then 🙂

**Comment by FlorianKempenich** [comment]
@lifeisafractal You've approved. I'm merging 👍

**Pull request merged**